# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1311714938 -7200
#      Tue Jul 26 23:15:38 2011 +0200
# Node ID 8d529f8103dc6265564b494962b32a715634914d
# Parent  5b168bbd6044a51e09ef44ebf960cecad8aa4d6d
Separated backend files to folder.

diff --git a/gui/octave-gui.pro b/gui/octave-gui.pro
--- a/gui/octave-gui.pro
+++ b/gui/octave-gui.pro
@@ -29,17 +29,17 @@ OBJECTS_DIR         = object-files      
 TRANSLATIONS        += languages/german             # Available translations
 LIBS                += -lqscintilla2
 
 mac {
     CONFIG -= app_bundle
 }
 
 # Includepaths and libraries to link against:
-INCLUDEPATH         += src src/terminal src/qirc
+INCLUDEPATH         += src src/terminal src/qirc src/backend
 INCFLAGS            += $$system(mkoctfile -p INCFLAGS)
 mac {
     INCFLAGS += -I/opt/local-native/include
 }
 
 LFLAGS              += $$system(mkoctfile -p LFLAGS) \
                        $$system(mkoctfile -p OCTAVE_LIBS) \
                        $$system(mkoctfile -p LIBS)
@@ -66,34 +66,34 @@ SOURCES +=\
         src/terminal/kpty.cpp \
         src/terminal/kptyprocess.cpp \
         src/terminal/kprocess.cpp \
         src/terminal/kptydevice.cpp \
         src/terminal/Session.cpp \
         src/terminal/ShellCommand.cpp \
         src/terminal/QTerminalWidget.cpp \
         src/MainWindow.cpp \
-        src/OctaveLink.cpp \
         src/terminal/ProcessInfo.cpp \
     	  src/OctaveTerminal.cpp \
     	  src/VariablesDockWidget.cpp \
     	  src/HistoryDockWidget.cpp \
     	  src/FilesDockWidget.cpp \
     	  src/FileEditorMdiSubWindow.cpp \
     	  src/BrowserWidget.cpp \
     	  src/ImageViewerMdiSubWindow.cpp \
     src/IRCWidget.cpp \
     src/qirc/IRCClient.cpp \
     src/qirc/IClientSocket.cpp \
     src/SettingsDialog.cpp \
     src/OctaveGUI.cpp \
     src/ResourceManager.cpp \
     src/CommandLineParser.cpp \
-    src/OctaveMainThread.cpp \
-    src/OctaveCallbackThread.cpp
+    src/backend/OctaveCallbackThread.cpp \
+    src/backend/OctaveLink.cpp \
+    src/backend/OctaveMainThread.cpp
 
 HEADERS += \
         src/terminal/TerminalCharacterDecoder.h \
         src/terminal/Character.h \
         src/terminal/CharacterColor.h \
 		  src/terminal/ColorTables.h \
         src/terminal/KeyboardTranslator.h \
         src/terminal/Screen.h \
@@ -112,29 +112,29 @@ HEADERS += \
         src/terminal/kptyprocess.h \
         src/terminal/kprocess.h \
         src/terminal/kprocess_p.h \
         src/terminal/kptydevice.h \
         src/terminal/Session.h \
         src/terminal/ShellCommand.h \
         src/terminal/QTerminalWidget.h \
     	  src/MainWindow.h \
-        src/OctaveLink.h \
         src/terminal/ProcessInfo.h \
     	  src/OctaveTerminal.h \
     	  src/VariablesDockWidget.h \
     	  src/HistoryDockWidget.h \
     	  src/FilesDockWidget.h \
     	  src/FileEditorMdiSubWindow.h \
     	  src/BrowserWidget.h \
     	  src/ImageViewerMdiSubWindow.h \
     src/IRCWidget.h \
     src/qirc/IRCCodes.h \
     src/qirc/IRCClient.h \
     src/qirc/IClientSocket.h \
     src/SettingsDialog.h \
     src/ResourceManager.h \
     src/CommandLineParser.h \
-    src/OctaveMainThread.h \
-    src/OctaveCallbackThread.h
+    src/backend/OctaveCallbackThread.h \
+    src/backend/OctaveLink.h \
+    src/backend/OctaveMainThread.h
 
 FORMS += \
     src/SettingsDialog.ui
diff --git a/gui/src/OctaveCallbackThread.cpp b/gui/src/backend/OctaveCallbackThread.cpp
rename from gui/src/OctaveCallbackThread.cpp
rename to gui/src/backend/OctaveCallbackThread.cpp
--- a/gui/src/OctaveCallbackThread.cpp
+++ b/gui/src/backend/OctaveCallbackThread.cpp
@@ -1,8 +1,26 @@
+/* OctaveGUI - A graphical user interface for Octave
+ * Copyright (C) 2011 Jacob Dawid
+ * jacob.dawid@googlemail.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "OctaveCallbackThread.h"
 #include "MainWindow.h"
 
 OctaveCallbackThread::OctaveCallbackThread (QObject * parent):QThread (parent)
 {
 }
 
 void
diff --git a/gui/src/OctaveCallbackThread.h b/gui/src/backend/OctaveCallbackThread.h
rename from gui/src/OctaveCallbackThread.h
rename to gui/src/backend/OctaveCallbackThread.h
--- a/gui/src/OctaveCallbackThread.h
+++ b/gui/src/backend/OctaveCallbackThread.h
@@ -1,8 +1,26 @@
+/* OctaveGUI - A graphical user interface for Octave
+ * Copyright (C) 2011 Jacob Dawid
+ * jacob.dawid@googlemail.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #ifndef OCTAVECALLBACKTHREAD_H
 #define OCTAVECALLBACKTHREAD_H
 
 #include <QThread>
 
 class OctaveCallbackThread:public QThread
 {
   Q_OBJECT
diff --git a/gui/src/OctaveLink.cpp b/gui/src/backend/OctaveLink.cpp
rename from gui/src/OctaveLink.cpp
rename to gui/src/backend/OctaveLink.cpp
diff --git a/gui/src/OctaveLink.h b/gui/src/backend/OctaveLink.h
rename from gui/src/OctaveLink.h
rename to gui/src/backend/OctaveLink.h
diff --git a/gui/src/OctaveMainThread.cpp b/gui/src/backend/OctaveMainThread.cpp
rename from gui/src/OctaveMainThread.cpp
rename to gui/src/backend/OctaveMainThread.cpp
--- a/gui/src/OctaveMainThread.cpp
+++ b/gui/src/backend/OctaveMainThread.cpp
@@ -1,8 +1,26 @@
+/* OctaveGUI - A graphical user interface for Octave
+ * Copyright (C) 2011 Jacob Dawid
+ * jacob.dawid@googlemail.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "OctaveMainThread.h"
 #include "OctaveLink.h"
 
 OctaveMainThread::OctaveMainThread (QObject * parent):QThread (parent)
 {
 }
 
 void
diff --git a/gui/src/OctaveMainThread.h b/gui/src/backend/OctaveMainThread.h
rename from gui/src/OctaveMainThread.h
rename to gui/src/backend/OctaveMainThread.h
--- a/gui/src/OctaveMainThread.h
+++ b/gui/src/backend/OctaveMainThread.h
@@ -1,8 +1,26 @@
+/* OctaveGUI - A graphical user interface for Octave
+ * Copyright (C) 2011 Jacob Dawid
+ * jacob.dawid@googlemail.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #ifndef OCTAVEMAINTHREAD_H
 #define OCTAVEMAINTHREAD_H
 
 #include <QThread>
 class OctaveMainThread:public QThread
 {
   Q_OBJECT
 public:
